apps/frontend/app/admin/_components/code-editor/EditorLayout.tsx

```diff
@@ -116,21 +115,66 @@ export function EditorLayout({
   const [isTesting, setIsTesting] = useState(false)
   const [testResults, setTestResults] = useState<TestResultDetail[]>([])
 
+  const initializeTestResults = useCallback(() => {
+    if (submissionData?.testcaseResult && testcaseData?.getProblem?.testcase) {
+      const submissionResultsMap = new Map(
+        submissionData.testcaseResult.map((r) => [
+          Number(r.problemTestcaseId),
+          r
+        ])
+      )
+      const problemTestcases = testcaseData.getProblem.testcase
+
+      let sampleCount = 0
```

Comment:
```suggestion
function mapTestResults(testcases, results) {
  const resultMap = new Map(results.map((r) => [Number(r.id), r]));
  let sampleCount = 0;
  let hiddenCount = 0;

  return testcases.map((testcase) => {
    const testResult = resultMap.get(Number(testcase.id));

    if (testcase.isHidden) {
      hiddenCount++;
    } else {
      sampleCount++;
    }

    return {
      id: Number(testcase.id),
      order: testcase.isHidden ? hiddenCount : sampleCount,
      type: testcase.isHidden ? 'Hidden' : 'Sample',
      input: testcase.input ?? '',
      expectedOutput: testcase.output ?? '',
      output: testResult?.output ?? '',
      result: testResult?.result ?? '',
      isUserTestcase: false,
    };
  });
}
```
initializeTestResults이랑 handleTest에서 겹치는 로직이 있어서 이렇게 함수를 파는게 어때요??

Reply:
반영했어용